pr921 Define cmsDriver step HGCalTriggerPrimitives

rekovic opened the pull request:
pr921 for PhaseII L1T MC processing.

Define step HGCalTriggerPrimitives to be used by cmsDriver command for PhaseII L1T MC reprocessing.

Reason:  
Since PhaseII MC samples were made, there was an important development in 
HGCalTriggerPrimitives available in CMSSW_9_2_1.
To be able to reprocess the samples centrally and run only the new HGCalTPs, need to define
it as a step to be used by `cmsDriver.py`

rekovic committed:
Define step HGCalTriggerPrimitives to be used by cmsDriver command for PhaseII L1T MC reprocessing.

rekovic commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by  @rekovic  for master.

It involves the following packages:

Configuration/Applications
Configuration/StandardSequences


@cmsbuild, @franzoni, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @makortel, @felicepantaleo, @GiacomoSguazzoni, @rovere, @VinInn, @Martin-Grunewald, @ebrondol, @dgulhan this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

kpedro88 commented:
assign upgrade

cmsbuild commented:
New categories assigned: upgrade

@kpedro88 you have been requested to review this Pull request/Issue and eventually sign? Thanks

kpedro88 commented on Configuration/Applications/python/ConfigBuilder.py:
```diff
@@ -1602,6 +1604,12 @@ def prepare_L1TrackTrigger(self, sequence = "L1TrackTrigger"):
+    def prepare_HGCalTriggerPrimitives(self, sequence = "HGCalTriggerPrimitives"):
+        ''' Enrich the schedule with L1 reconstruction '''
+        self.loadDefaultOrSpecifiedCFF(sequence,self.HGCalTriggerPrimitivesDefaultCFF)
+	self.scheduleSequence(sequence.split('.')[-1],'HGCalTriggerPrimitives_step')
```
please use consistent indentation in python code (don't mix tabs and spaces)

rekovic commented:
@kpedro88, @davidlange6  
As discussed at the ORP, due to changed geometry, might have to go back to 90X,
in which case we would need 90x backport of HGCal TPs PR https://github.com/cms-sw/cmssw/pull/18861.

kpedro88 commented:
@jbsauvan FYI

Does #18861 work for both HGCal geometries (v7 and v8)?

rekovic committed:
Uniform indentation in python file.

cmsbuild commented:
Pull request #19115 was updated. @cmsbuild, @franzoni, @kpedro88, @davidlange6 can you please check and sign again.

rekovic commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

jbsauvan commented:
There are still missing pieces for the V8 geometry, so the HGCAL TPG won't work with V8. I'm trying to get  everything working with V8 this week.

kpedro88 commented:
okay, that's actually better for this purpose - L1 TDR samples were generated with 2023D4, which has v7 HGCal.

cmsbuild commented:
+1
Tested at: f4632c794fc12520bdad236d480033dcd9b0e4d2
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19115/20374/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19115/20374/summary.html

Comparison Summary:
 - You potentially added 36 lines to the logs
 - Reco comparison results: 3422 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1837639
 - DQMHistoTests: Total failures: 18214
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1819252
 - DQMHistoTests: Total skipped: 173
 - DQMHistoTests: Total Missing objects: 0
 - Checked 94 log files, 14 edm output root files, 23 DQM output files

rekovic commented:
@kpedro88 
From the response of @jbsauvan, I was to understand that we were OK to use it on v7 geometry.

- In CMSSW_920 + rebased #18861 + rebased this PR,
I tried to run this step `-s HGCalTriggerPrimitives` on L1T MC samples and it runs fine.
Example command:
`cmsDriver.py step2 --python_filename=test_rerun_step2_HGCalTPs_onMCL1_FEVTHLTDEBUG.py --no_exec -s HGCalTriggerPrimitives --datatier GEN-SIM-DIGI-RAW -n 5 --era Phase2C2 --eventcontent FEVTDEBUGHLT --pileup AVE_200_BX_25ns --filein root://cms-xrd-global.cern.ch///store/mc/PhaseIISpring17D/SingleNeutrino/GEN-SIM-DIGI-RAW/PU200_90X_upgrade2023_realistic_v9-v1/70000/004C6E8B-5F26-E711-AE46-0242AC130002.root --conditions 90X_upgrade2023_realistic_v1 --era Phase2C2_timing --beamspot HLLHC14TeV --geometry Extended2023D4 --fileout file:step2_PU200_rerun-HGCalTPs.root
`

- In CMSSW_921 + this PR, the same command above fails
<pre>
----- Begin Fatal Exception 07-Jun-2017 10:36:38 CEST-----------------------
An exception of category 'NoProxyException' occurred while
   [0] Processing  stream begin Run run: 1 stream: 0
   [1] Calling method for module HGCalTriggerDigiProducer/'hgcalTriggerPrimitiveDigiProducer'
   [2] Using EventSetup component HGCalTriggerGeometryESProducer/'hgcalTriggerGeometryESProducer' to make data HGCalTriggerGeometryBase/'' in record IdealGeometryRecord
Exception Message:
No data of type "HGCalTopology" with label "HGCalEESensitive" in record "IdealGeometryRecord"
 Please add an ESSource or ESProducer to your job which can deliver this data.
----- End Fatal Exception -------------------------------------------------
</pre>

Addtiional note:  need to add all products "HGCalTriggerPrimitives" to EventContent FEVTDEBUGHLT.

rekovic committed:
Add products from hgcalTriggerPrimitiveDigiProducer to EventContent FEVTDEBUG.

cmsbuild commented:
Pull request #19115 was updated. @civanch, @mdhildreth, @cmsbuild, @franzoni, @kpedro88, @davidlange6 can you please check and sign again.

rekovic commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 9f0f46e77a5f867192cb0112a91fddc694254d96
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19115/20395/summary.html

cmsbuild commented:
Comparison job queued.

kpedro88 commented:
@rekovic The 2023D4 geometry no longer exists in 9_2_1 - onward. This is why the PRs need to be backported to 90X for compatibility with the L1 TDR samples.

davidlange6 commented:
in any case the usual model for trigger primitives is that they are defined as part of other steps (remembering that you can use a syntax like L1:hgcalTriggerPrimitives to run an alternative sequence) . putting these changes into ConfigBuilder is not the right place for them.

rekovic commented:
@davidlange6 
>@rekovic The 2023D4 geometry no longer exists in 9_2_1 - onward. This is why the PRs need to be backported to 90X for compatibility with the L1 TDR samples.

Since this runs fine on top of 9_2_0, a patch to 9_2_0 would be fine?
PRs (this one + rebased #18861) need to be made to which branch?  9_2_0_patchX ?

kpedro88 commented:
@rekovic there have been numerous other changes since 9_0_X that could cause inconsistencies...

davidlange6 commented:
we need a master PR and then a 90x PR it seems.

> On Jun 7, 2017, at 2:59 PM, rekovic <notifications@github.com> wrote:
> 
> @davidlange6
> 
> @rekovic The 2023D4 geometry no longer exists in 9_2_1 - onward. This is why the PRs need to be backported to 90X for compatibility with the L1 TDR samples.
> 
> Since this runs fine on top of 9_2_0, a patch to 9_2_0 would be fine?
> PRs (this one + rebased #18861) need to be made to which branch? 9_2_0_patchX ?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

rekovic commented:
@kpedro88 @davidlange6 
OK.  Understood.  Need PRs to 90x.  Thanks.

rekovic committed:
Add hgcalTriggerPrimitives to phase2_ecalDigiSequence.

rekovic committed:
Revert "Uniform indentation in python file."

This reverts commit f4632c794fc12520bdad236d480033dcd9b0e4d2.

rekovic committed:
Revert "Define step HGCalTriggerPrimitives to be used by cmsDriver command for PhaseII L1T MC reprocessing."

This reverts commit 03c4d6264e0db457cc7372eada989fbff5253592.

cmsbuild commented:
Pull request #19115 was updated. @cmsbuild, @civanch, @mdhildreth, @kpedro88, @davidlange6 can you please check and sign again.

rekovic commented:
@davidlange6 
>in any case the usual model for trigger primitives is that they are defined as part of other steps (remembering that you can use a syntax like L1:hgcalTriggerPrimitives to run an alternative sequence) . putting these changes into ConfigBuilder is not the right place for them.

Ok.  Added hgcalTriggerPrimitives to phase2_ecalDigiSequence, and reverted the changes to ConfigBulider.  This step `-s DIGI:hgcalTriggerPrimitives` runs fine and correct products are saved, in CMSSW_9_2_0 with command

`cmsDriver.py step2 --python_filename=test_rerun_step2_HGCalTPs_onMCL1_FEVTHLTDEBUG.py --no_exec -s DIGI: --datatier GEN-SIM-DIGI-RAW -n 5 --era Phase2C2 --eventcontent FEVTDEBUGHLT --pileup AVE_200_BX_25ns --filein root://cms-xrd-global.cern.ch///store/mc/PhaseIISpring17D/SingleNeutrino/GEN-SIM-DIGI-RAW/PU200_90X_upgrade2023_realistic_v9-v1/70000/004C6E8B-5F26-E711-AE46-0242AC130002.root --conditions 90X_upgrade2023_realistic_v1 --era Phase2C2_timing --beamspot HLLHC14TeV --geometry Extended2023D4 --fileout file:step2_PU200_rerun-HGCalTPs.root
`

rekovic commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

kpedro88 commented:
1) `ecalDigiSequence` is not the right place for `hgcalTriggerPrimitives`. These are entirely separate subdetectors.
2) Rather than reverting the previous commits and leaving a messy history, just rebase them out of the branch entirely.

cmsbuild commented:
-1

Tested at: 246518d489734e0c20c3204a346dc2eeba77f6a8

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19115/20403/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
20434.0 step2
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D10_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D10+RecoFullGlobal_2023D10+HARVESTFullGlobal_2023D10/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D10_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D10+RecoFullGlobal_2023D10+HARVESTFullGlobal_2023D10.log

</pre>27434.0 step2
<pre>runTheMatrix-results/27434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFull_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFull_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

rekovic commented:
@kpedro88 
>1. `ecalDigiSequence` is not the right place for `hgcalTriggerPrimitives`. These are entirely separate subdetectors.

>2. Rather than reverting the previous commits and leaving a messy history, just rebase them out of the branch entirely.
1) You are right of course.  A proper thing to do is to add `SimCalorimetry/Configuration/python/hgcalDigiSequence_cff.py`.  will do.
2) Indeed, can make a PR with clean history once arrived to proper solution.

kpedro88 commented:
@rekovic what is the status of this PR?
